### PR TITLE
Feature/117 import xml json from files

### PR DESCRIPTION
**Pull request by sastafford**

### CODE REVIEW

File: core/src/main/java/com/marklogic/spring/batch/item/MarkLogicFileItemWriter.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.List;
+
+@Component
+@Scope("step")
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/item/MarkLogicFileItemWriter.java]: I wouldn't put annotations here, I think it constrains usage of the writer. I think in general, we want Config classes to use annotations to scope objects. Readers/writers/processors should largely be unaware of the scope that they'll be used in.

File: core/src/main/java/com/marklogic/spring/batch/item/MarkLogicImportItemProcessor.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    public FileHandle process(Resource item) throws Exception {
+        File file = item.getFile();
+        FileHandle handle = new FileHandle(file);
+        String fileName = file.getName();
+        int i = fileName.lastIndexOf('.');
+        String extension = (i >= 0) ? fileName.substring(i+1) : ".xyz";
+        if (format.equals(Format.XML) || extension.equals("xml")) {
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/item/MarkLogicImportItemProcessor.java]: I'd reverse the equals check so there's never a chance of a null-pointer error on format - e.g. if (Format.XML.equals(format))

File: core/src/main/java/com/marklogic/spring/batch/item/MarkLogicFileItemWriter.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public void open(ExecutionContext executionContext) {
+        this.executionContext = executionContext;
+        docMgr = client.newDocumentManager();
+    }
+
+    @Override
+    public void close() {
+        client.release();
```
Comment (rjrudin) [inline core/src/main/java/com/marklogic/spring/batch/item/MarkLogicFileItemWriter.java]: I don't think you need to release the client. The Writer didn't open it, so it shouldn't close it - there might be other classes using it. The expectation is that the DatabaseClient was instantiated via Spring, via a class like DatabaseClientManager, which will release the client when the Spring container shuts down.

File: jobs/file-importers/src/main/java/com/marklogic/spring/batch/config/ImportDocumentsFromDirectoryConfig.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    @JobScope
+    protected Step step(
+        @Value("#{jobParameters['input_file_path']}") String inputFilePath,
+        @Value("#{jobParameters['input_file_pattern']}") String inputFilePattern,
+        @Value("#{jobParameters['document_type']}") String documentType,
+        @Value("#{jobParameters['output_uri_prefix']}") String outputUriPrefix,
+        @Value("#{jobParameters['output_uri_replace']}") String outputUriReplace,
+        @Value("#{jobParameters['output_uri_suffix']}") String outputUriSuffix,
+        @Value("#{jobParameters['output_collections']}") String outputCollections,
+        @Value("#{jobParameters['chunk'] ?: 100}") Integer chunkSize) {
```
Comment (rjrudin) [inline jobs/file-importers/src/main/java/com/marklogic/spring/batch/config/ImportDocumentsFromDirectoryConfig.java]: No need to get chunk here - the parent class already has a getChunkSize() method that you can use. It checks the Environment for the chunk parameter, and defaults to 10 if it's not set .

File: jobs/file-importers/src/main/java/com/marklogic/spring/batch/config/ImportDocumentsFromDirectoryConfig.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+public class ImportDocumentsFromDirectoryConfig extends AbstractMarkLogicBatchConfig {
+
+    @Bean
+    public Job job(Step step) {
+        return jobBuilderFactory.get("importDocumentsFromDirectoryJob").start(step).build();
+    }
+
+    @Bean
+    @JobScope
+    protected Step step(
```
Comment (rjrudin) [inline jobs/file-importers/src/main/java/com/marklogic/spring/batch/config/ImportDocumentsFromDirectoryConfig.java]: I like this - I think it's nice to see in one place all the job parameters supported by a step. I don't think there's any downside to having a ton of parameters here, as no one but Spring ever invokes this method.

**Comment by sastafford** [comment]
This issue will close both #60 and #108.  The URI generation is still a bit of work in progress.  What I have discovered that the URI generation is going to happen in the ItemWriter.  URI generation is dependent on the batch process job.  But we should utilize the URIGeneration interface.

**Comment by sastafford** [comment]
Of course this also closes #117

**Pull request merged**